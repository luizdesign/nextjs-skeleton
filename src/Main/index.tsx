const Main = () => (
  <main>
    <h1>NextJs Skeleton</h1>
  </main>
);

export default Main;
