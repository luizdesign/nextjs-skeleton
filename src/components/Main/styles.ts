import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Logo = styled.img`
  width: 15rem;
`;

export const Title = styled.h1`
  margin: 5px 0 10px;
  font-size: 3rem;
`;

export const Description = styled.strong`
  font-size: 1.5rem;
`;
