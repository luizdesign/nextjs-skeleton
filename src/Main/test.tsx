import { screen, render } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('Should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', {
        name: /NextJs Skeleton/,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
