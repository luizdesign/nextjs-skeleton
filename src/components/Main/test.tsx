import { screen, render } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('Should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', {
        name: /NextJs Skeleton/,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render the style correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b',
    });
  });
});
