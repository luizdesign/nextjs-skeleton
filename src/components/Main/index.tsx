import * as S from './styles';

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um Ã¡tomo e NextJs Skeleton escrito ao lado"
    />
    <S.Title>NextJs Skeleton</S.Title>
    <S.Description>
      NextJs, Typescript, ReactJs e Styled Components
    </S.Description>
  </S.Wrapper>
);

export default Main;
