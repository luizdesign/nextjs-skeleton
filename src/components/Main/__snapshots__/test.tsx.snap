// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-1fpvr54-0 ghaTmi"
>
  <img
    alt="Imagem de um Ã¡tomo e NextJs Skeleton escrito ao lado"
    class="styles__Logo-sc-1fpvr54-1 iJIHgf"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1fpvr54-2 kKQKnq"
  >
    NextJs Skeleton
  </h1>
  <strong
    class="styles__Description-sc-1fpvr54-3 ixQzoP"
  >
    NextJs, Typescript, ReactJs e Styled Components
  </strong>
</main>
`;
